runtime/lua/vim/lsp.lua

```diff
@@ -209,17 +209,117 @@ local function reuse_client_default(client, config)
   return false
 end
 
+--- Reset defaults set by `set_defaults`.
+--- Must only be called if the last client attached to a buffer exits.
+local function reset_defaults(bufnr)
```

Comment:
Sorry for the large reordering diff here. Needed to move stuff around to make `create_and_initialize_client` a local function.